File: treinamentoP1/combinacao.c
/* Solução teoricamente possivel Mas,
 inviavel na pratica */

#include <stdio.h>

double comb(double n, double p);

int main(void)
{
	double n,p;
	scanf("%lf %lf", &n, &p);
	
	printf("%f\n", comb(n,p) );
	return 0;
}

double comb(double n, double p)
{
	if( p == 0)
		return 1.0;
	else{
		return  comb(n,p-1) * (n-(p-1)) / p;
	}
}

File: treinamentoP1/combinacao.py
# Solução teoricamente possivel Mas,
# inviavel na pratica


def comb(n,k):
	if(n==k):
		return 1
	else:
		return comb(n, k-1) * (1/k*(n-k))

print( comb(10,3) )
